pkg/config/template.go

```diff
@@ -0,0 +1,31 @@
+package config
+
+import (
+	"bytes"
+	"fmt"
+	"text/template"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func TemplateLabels(labels map[string]string, target *unstructured.Unstructured) (map[string]string, error) {
```

Comment:
Does it really belong to `config` package?

I am also not sure if it shouldn't be called `ResolveLabelsValues` or sth between the lines and second argument is rather a tmpl data or source?

Some godoc would help too, especially what you capture in `CAUTION` part in the PR description.

Reply:
Not sure which package it belongs to to be honest. It went into config because that's the source of the data to "resolve". 

Renamed ResolveSelectors, also added resolve of Key as well as Value and a check if the key/value contain any expression brackets to avoid running pure text through the template engine for no reason.